modules/core/src/mathfuncs_core.simd.hpp

```diff
@@ -185,13 +110,60 @@ static inline void atanImpl(const T *Y, const T *X, T *angle, int len, bool angl
 void fastAtan32f(const float *Y, const float *X, float *angle, int len, bool angleInDegrees )
 {
     CV_INSTRUMENT_REGION()
-    atanImpl<float>(Y, X, angle, len, angleInDegrees);
+    float scale = angleInDegrees ? 1.f : (float)(CV_PI/180);
+    int i = 0;
+#if CV_SIMD
+    const int VECSZ = v_float32::nlanes;
+    v_atan_f32 v(scale);
+
+    for( ; i < len; i += VECSZ*2 )
+    {
+        if( i + VECSZ*2 > len )
+        {
+            // if it's inplace operation, we cannot repeatedly process
+            // the tail for the second time, so we have to use the
+            // scalar code
+            if( i == 0 || angle == X || angle == Y )
+                break;
+            i = len - VECSZ*2;
+        }
+
+        v_float32 y0 = vx_load(Y + i);
+        v_float32 x0 = vx_load(X + i);
+        v_float32 y1 = vx_load(Y + i + VECSZ);
+        v_float32 x1 = vx_load(X + i + VECSZ);
+
+        v_float32 r0 = v.compute(y0, x0);
+        v_float32 r1 = v.compute(y1, x1);
+
+        v_store(angle + i, r0);
+        v_store(angle + i + VECSZ, r1);
+    }
+    vx_cleanup();
+#endif
+
+    for( ; i < len; i++ )
+        angle[i] = atan_f32(Y[i], X[i])*scale;
 }
 
 void fastAtan64f(const double *Y, const double *X, double *angle, int len, bool angleInDegrees)
 {
     CV_INSTRUMENT_REGION()
-    atanImpl<double>(Y, X, angle, len, angleInDegrees);
+
+    const int BLKSZ = 128;
+    float ybuf[BLKSZ], xbuf[BLKSZ], abuf[BLKSZ];
+    for( int i = 0; i < len; i += BLKSZ )
+    {
+        int j, blksz = std::min(BLKSZ, len - i);
+        for( j = 0; j < blksz; j++ )
+        {
+            ybuf[j] = (float)Y[i + j];
+            xbuf[j] = (float)X[i + j];
+        }
+        fastAtan32f(ybuf, xbuf, abuf, blksz, angleInDegrees);
```

Comment:
`fastAtan32f()` is function with `CV_INSTRUMENT_REGION()` (required for OpenCV entry-points, but using this inside of "per-element" loops should be avoided).

Performance may be terrible, than computation in doubles.
Do we have performance tests for this?